docs/tutorial/7-snapping.md

```diff
@@ -0,0 +1,610 @@
+
+## Installing requirements
+
+Some dependencies are required when building snap packages:
+
+- Confirm that you have [snapd][snapd] installed by running the `snap version` command.
+  If you're using Ubuntu, it should come pre-installed. 
+  Otherwise, refer to the [snapd installation docs][install_snapd].
+
+- Install the [snapcraft snap][snapcraft_docs]:
+  ```shell
+    sudo snap install --classic snapcraft
+  ```
+
+## Packaging
+
+It is easier to create the snap by starting from a template.
+Run the following command from the same directory where the source `.c` files reside:
+
+```{terminal}
+   :input: snapcraft init
+   :user: ubuntu
+   :host: ubuntu
+   :dir: ~/tutorial
+
+Go to https://docs.snapcraft.io/the-snapcraft-format/8337 for more information about the snapcraft.yaml format.
+Successfully initialised project.
+```
+This command creates a `snap` directory containing the `snapcraft.yaml` file.
+
+
+Let's examine the `snapcraft.yaml` file:
+
+```yaml
+name: demo # you probably want to 'snapcraft register <name>'
+base: core24 # the base snap is the execution environment for this snap
+version: '0.1' # just for humans, typically '1.2+git' or '1.3.2'
+summary: Single-line elevator pitch for your amazing snap # 79 char long summary
+description: |
+  This is my-snap's description. You have a paragraph or two to tell the
+  most important story about your snap. Keep it under 100 words though,
+  we live in tweetspace and your description wants to look good in the snap
+  store.
+
+grade: devel # must be 'stable' to release into candidate/stable channels
+confinement: devmode # use 'strict' once you have the right plugs and slots
+
+parts:
+  my-part:
+    # See 'snapcraft plugins'
+    plugin: nil
+```
+You may also want to review the [snapcraft.yaml schema][snapcraft_yaml], which is the reference for all the possible configuration keys for the `snapcraft.yaml` file.
+
+Your working directory should now look like this:
+
+```{terminal}
+   :input: tree
+   :user: ubuntu
+   :host: ubuntu
+   :dir: ~/tutorial
+
+.
+├── cfs.c
+├── edf.c
+├── fifo.c
+├── lock.c
+├── snap
+│   └── snapcraft.yaml
+└── thread-affinity.c
+
+2 directories, 6 files
+```
+
+
+
+We'll also make some modifications to the `snapcraft.yaml` file:
+
+- Change the name from **demo** to **rt-app**.
+- Provide a meaningful summary and description of what the snap does.
+
+### Defining the snap parts
+
+Once the Snap metadata boilerplate is in place, it's time to define the [snapcraft parts][snapcraft_parts].
+In Snaps, `parts` serve as the building blocks of the package, similar to a recipe's list of ingredients and preparation steps.
+
+Since we have five `.c` source files, we might assume that we need five separate parts.
+While that approach is possible, it's much easier to create a Makefile and use that to build all the program.
+Let's create a Makefile for our real-time apps.
+Adding the build logic to a Makefile simplifies the packaging.
+Let's create one:
```

Comment:
Are lines 94-95 left here by mistake?

```suggestion
Since we have five `.c` source files, we might assume that we need five separate parts.
While that approach is possible, it's much easier to create a Makefile and use that to build all the programs.
Let's create a Makefile for our real-time apps:
```